src/EditorFeatures/CSharpTest/ConvertForEachToFor/ConvertForEachToForTests.cs

```diff
@@ -1721,5 +1721,23 @@ class Explicit : IReadOnlyList<int>
 ";
             await TestInRegularAndScriptAsync(text, expected, options: ImplicitTypeEverywhere);
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertForEachToFor)]
+        public async Task ArrayRank2()
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        foreach [||] (int a in new int[,] { {1, 2} })
```

Comment:
note: this is somewhat dangerous.  If we ever changed the span to be only teh 'foreach', then this test would continue failing, but not for the right reason.  Code action tests try to pick a span that should always be safe even in the presence of future refinements to the applicability span.